nixos/modules/services/mail/dovecot.nix

```
@@ -7,7 +7,94 @@ let
   dovecotPkg = pkgs.dovecot;
 
   baseDir = "/run/dovecot2";
-  stateDir = "/var/lib/dovecot";
+
+  sieveToPath = text:
+    # Sieve scripts can start with a block comment
+    if isString text && hasPrefix "/*" text
+      then pkgs.writeText "script.sieve" text
+    # Paths in the store are used as is
+    else if types.path.check text && hasPrefix "${builtins.storeDir}/" (toString text)
+      then text
+    # Paths outside the store get written to the store
+    else if types.path.check text
+      then pkgs.writeText "script.sieve" (fileContents text)
+    # All other is assumed to be file content
+    else pkgs.writeText "script.sieve" text;
+  scriptType = types.coercedTo types.lines sieveToPath types.path;
+
+  compileSieveScripts = pkgs.buildSieveScripts {
```

```suggestion
  compiledSieveScripts = pkgs.buildSieveScripts {
```

Does compiling these in advance break `include` for non-store paths? Someone may be using that for dynamically generated rules. If it does, would they be doable another way?